code/controllers/subsystem/mapping.dm

```diff
@@ -533,9 +533,11 @@ GLOBAL_LIST_EMPTY(the_station_areas)
 
 /datum/controller/subsystem/mapping/proc/preloadRuinTemplates()
 	// Still supporting bans by filename
-	var/list/banned = generateMapList("lavaruinblacklist.txt")
-	banned += generateMapList("spaceruinblacklist.txt")
-	banned += generateMapList("iceruinblacklist.txt")
+	var/list/banned = generateMapList("spaceruinblacklist.txt")
+	if(config.minetype == "lavaland")
```

Comment:
Make this a switch/case instead so a problem like this doesn't happen again in the future

Reply:
problem is that icebox doesn't have a minetype. There's only 'lavaland' and 'none', because icebox's icemoon is literally just the station itself

Reply:
That's...pretty bad.

We need some way for this to be forwards compatible for if we add a new kind of mining station. Since minetype is just a part of map config, what would break if we had Ice Box specify a different type? If it's not possible for whatever reason, maybe we should enforce that minetype = none also requires you specify the blacklist file in your configuration?

Reply:
We could specify any type we want but for the case of icebox it would just be "none 2"
blacklist file would probably be best

Reply:
I changed it, though I was thinking of maybe only using the blacklist_file, that way I suppose any map (downstream or not) who would want to allow different types of lavaland spawners unique to their own maps would be allowed, but 🤷